internal/characters/kazuha/skill.go

```diff
@@ -7,50 +7,60 @@ import (
 	"github.com/genshinsim/gcsim/pkg/core/combat"
 )
 
-var skillPressFrames []int
-var skillHoldFrames []int
+var skillPressFrames [][]int
+var skillHoldFrames [][]int
 
 const (
 	skillPressHitmark = 10
+	skillPressRadius  = 5
 	skillPressCDStart = 8
 	skillHoldHitmark  = 33
+	skillHoldRadius   = 9
 	skillHoldCDStart  = 31
 	particleICDKey    = "kazuha-particle-icd"
 )
 
 func init() {
-	//TODO: glide cancel
-	// skill (press) -> x
-	//85 frames to float down
-	skillPressFrames = frames.InitAbilSlice(77) //averaged all abils
-	//27 frames before the start of plunge animation
-	skillPressFrames[action.ActionHighPlunge] = 24
-
-	// skill (hold) -> x
-	//177 frames to float down
-	skillHoldFrames = frames.InitAbilSlice(175) //averaged all abils
-	//58 frames before start of plunge animation
-	skillHoldFrames[action.ActionHighPlunge] = 58
+	// Tap E
+	skillPressFrames = make([][]int, 2)
+	// Tap E -> X
+	skillPressFrames[0] = frames.InitAbilSlice(77) // averaged all abils
+	skillPressFrames[0][action.ActionHighPlunge] = 24
+	// Tap E (Glide Cancel) -> X
+	skillPressFrames[1] = frames.InitAbilSlice(69)
+	skillPressFrames[1][action.ActionBurst] = 61
+	skillPressFrames[1][action.ActionDash] = 61
+	skillPressFrames[1][action.ActionJump] = 59
+	skillPressFrames[1][action.ActionSwap] = 60
+
+	// Hold E
+	skillHoldFrames = make([][]int, 2)
+	// Hold E -> X
+	skillHoldFrames[0] = frames.InitAbilSlice(175) // averaged all abils
+	skillHoldFrames[0][action.ActionHighPlunge] = 58
+	// Hold E (Glide Cancel) -> X
+	skillHoldFrames[1] = frames.InitAbilSlice(160)
+	skillHoldFrames[1][action.ActionAttack] = 158
+	skillHoldFrames[1][action.ActionBurst] = 159
+	skillHoldFrames[1][action.ActionSwap] = 155
 }
 
 func (c *char) Skill(p map[string]int) action.ActionInfo {
 	hold := p["hold"]
-
-	c.a1Absorb = attributes.NoElement
-
-	radius := 5.0
-
-	if hold >= 1 {
-		radius = 9
+	glide := p["glide"]
+	if glide < 0 {
+		glide = 0
+	}
+	if glide > 1 {
+		glide = 1
 	}
```

Comment:
nit: `glide_cancel` or similar for param name to make it clear that this is a glide cancel and not just gliding in sim (my impression of `glide` would be acting like a wait between the skill and plunge where you hover in air)